docs/examples/basic_usage/benchmark_utils.py

```diff
@@ -0,0 +1,146 @@
+from torchvision.models import ResNet50_Weights, resnet50
+
+# In the execution with sphinx-gallery, __file__ is not defined and we need
+# to set it manually using the trick from https://stackoverflow.com/a/53293924
+if "__file__" not in globals():
+    __file__ = inspect.getfile(lambda: None)
+
+HEREDIR = path.dirname(path.abspath(__file__))
+
+
+def maybe_download_nanogpt_shakespeare() -> str:
+    """Download the nanoGPT model and Shakespeare data pre-processing script and run it.
+
+    Only performs the download and pre-processing if the data does not exist.
+
+    Returns:
+        The directory containing the Shakespeare data.
+    """
+    commit = "f08abb45bd2285627d17da16daea14dda7e7253e"
+    repo = "https://raw.githubusercontent.com/karpathy/nanoGPT/"
+
+    # download the model definition as 'nanogpt_model.py'
+    model_url = f"{repo}{commit}/model.py"
+    model_path = path.join(HEREDIR, "nanogpt_model.py")
+    if not path.exists(model_path):
+        url = requests.get(model_url)
+        with open(model_path, "w") as f:
+            f.write(url.content.decode("utf-8"))
+
+    # download the data pre-processing script from Shakespeare and execute it
+    data_url = f"{repo}{commit}/data/shakespeare/prepare.py"
+    data_dir = path.join(HEREDIR, "shakespeare")
+    makedirs(data_dir, exist_ok=True)
+    data_path = path.join(data_dir, "prepare.py")
+    if not path.exists(data_path):
+        url = requests.get(data_url)
+        with open(data_path, "w") as f:
+            f.write(url.content.decode("utf-8"))
+        run(["python", data_path])
+
+    return data_dir
+
+
+class GPTWrapper(Module):
+    """Wraps Karpathy's nanoGPT model repo so that it produces the flattened logits."""
+
+    def __init__(self, gpt: Module):
+        """Store the wrapped nanoGPT model.
+
+        Args:
+            gpt: The nanoGPT model.
+        """
+        super().__init__()
+        self.gpt = gpt
+
+    def forward(self, x: Tensor) -> Tensor:
+        """Forward pass of the nanoGPT model.
+
+        Args:
+            x: The input tensor. Has shape ``(batch_size, sequence_length)``.
+
+        Returns:
+            The flattened logits.
+            Has shape ``(batch_size * sequence_length, vocab_size)``.
+        """
+        y_dummy = zeros_like(x)
+        logits, _ = self.gpt(x, y_dummy)
+        return logits.view(-1, logits.size(-1))
+
+
+def setup_shakespeare_nanogpt(
+    batch_size: int = 2,
+) -> Tuple[GPTWrapper, CrossEntropyLoss, List[Tuple[Tensor, Tensor]]]:
+    """Set up the nanoGPT model and Shakespeare dataset for the benchmark.
+
+    Args:
+        batch_size: The batch size to use. Default is ``2``.
+
+    Returns:
+        A tuple containing the nanoGPT model, the loss function, and the data.
+    """
+    # download nanogpt_model and import GPT and GPTConfig from it
+    # also download the Shakespeare dataset
+    data_dir = maybe_download_nanogpt_shakespeare()
```

Comment:
Is there a reason you use real instead of synthetic data here?

Reply:
No. I rather started with the real data set to avoid any issues when adding the GPT model. Will refactor to use synthetic data instead.